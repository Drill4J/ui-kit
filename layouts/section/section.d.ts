/// <reference types="react" />
interface Props {
    className?: string;
    children: React.ReactNode;
    header: React.ReactNode;
}
export declare const Section: ({ className, header, children }: Props) => JSX.Element;
export {};
