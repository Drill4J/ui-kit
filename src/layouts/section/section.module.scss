@import '../../style-guide/index';
.section {
  display: flex;
  flex-direction: column;

  &__header {
    display: flex;
    align-items: center;
    height: 56px;
    font-size: 16px;
    line-height: 24px;
    color: map-get($color-palette, black100);
    border-bottom: 1px solid map-get($color-palette, light-blue);
  }
}
