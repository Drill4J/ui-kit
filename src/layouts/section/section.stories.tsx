import React from 'react';
import { storiesOf } from '@storybook/react';

import { Section } from './section';

storiesOf('Section', module).add('Section', () => (
  <Section header={<h1>Main</h1>}>
    <div>Section</div>
    <Section header={<h2>Secondary</h2>}>
      <span>Section</span>
    </Section>
  </Section>
));
