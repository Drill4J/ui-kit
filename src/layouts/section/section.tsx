import { BEM } from '@redneckz/react-bem-helper';

import styles from './section.module.scss';

interface Props {
  className?: string;
  children: React.ReactNode;
  header: React.ReactNode;
}

const section = BEM(styles);

export const Section = section(({ className, header, children }: Props) => (
  <div className={className}>
    <Header>{header}</Header>
    {children}
  </div>
));

const Header = section.header('div');
