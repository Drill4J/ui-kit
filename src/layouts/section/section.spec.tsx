import React from 'react';
import renderer from 'react-test-renderer';

import { Section } from './section';

it('should match snapshot', () => {
  const tree = renderer
    .create(
      <Section header={<h1>Main</h1>}>
        <div>Section</div>
        <Section header={<h2>Secondary</h2>}>
          <span>Section</span>
        </Section>
      </Section>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
