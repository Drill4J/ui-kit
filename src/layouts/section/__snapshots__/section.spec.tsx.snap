// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="section"
>
  <div
    className="section__header"
    data-modifiers=""
  >
    <h1>
      Main
    </h1>
  </div>
  <div>
    Section
  </div>
  <div
    className="section"
  >
    <div
      className="section__header"
      data-modifiers=""
    >
      <h2>
        Secondary
      </h2>
    </div>
    <span>
      Section
    </span>
  </div>
</div>
`;
